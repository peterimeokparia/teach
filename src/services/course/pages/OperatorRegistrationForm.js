import 
React, { 
useState } from 'react';

import './LoginPage.css';
 

   

const OperatorRegistrationForm = ({ 
  error, 
  loading,
  handleCreateUser }) => {

  const [ email, setEmail ] = useState('');
  const [ password, setPassword ] = useState('');
  const [ firstName, setFirstName ] = useState('');
  const [ lastName, setLastName ] = useState('');
  const [ businessName, setBusinessName ] = useState('');
  const [ phone, setPhone ] = useState('');




   const handleSubmit = (e) => {
      e.preventDefault();
   }


  
  return   (    
            <div className="LoginPage"> 

              <form onSubmit={ e => handleSubmit(e)}>
       
                  <label>  
                 
                    Email

                    <input
                        name="email"
                        type="email"
                        value={email}
                        onChange={ e => setEmail( e.target.value ) }
                        placeholder="email"
                    >
                    </input>
                  </label>


                   <label>

                    Password   

                    <input
                        name="password"
                        type="password"
                        value={password}
                        onChange={ e => setPassword( e.target.value ) }
                        placeholder="password"
                    >
                    </input>
                  </label>


                  <label>

                    First Name   

                    <input
                        name="firstName"
                        type="text"
                        value={firstName}
                        onChange={ e => setFirstName( e.target.value ) }
                        placeholder="first name"
                    >
                    </input>
                  </label>

                  <label>

                    Last Name   

                    <input
                        name="lastName"
                        type="text"
                        value={lastName}
                        onChange={ e => setLastName( e.target.value ) }
                        placeholder="last name"
                    >
                    </input>
                    </label>


                    <label>

                        Business Name   

                        <input
                            name="businessName"
                            type="text"
                            value={businessName}
                            onChange={ e => setBusinessName( e.target.value ) }
                            placeholder="business name"
                        >
                        </input>
                        </label>

                        <label>

                          Phone

                          <input
                              name="phone"
                              type="phone"
                              value={phone}
                              onChange={ e => setPhone( e.target.value ) }
                              placeholder="phone"
                          >
                          </input>
                          </label>

    
                

                  <div>
                                  
                  </div>

                    { error  && (<div className="error"> { error.message }</div>)}

                     <div></div>
                     
                        <button
                             type="submit"
                             disabled={loading}
                             onClick={e => handleCreateUser(email, password, firstName, lastName, businessName, phone)}
                         >
   
                            Create Service Account
   
                         </button>
               </form> 

               {/* {Validations.setErrorMessageContainer()} */}
                       
            </div> 
    );

}



export default OperatorRegistrationForm;