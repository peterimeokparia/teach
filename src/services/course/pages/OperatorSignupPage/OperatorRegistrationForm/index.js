import { 
useState } from 'react';

import {
role } from 'services/course/helpers/PageHelpers';

import './style.css';

const OperatorRegistrationForm = ({ 
  error, 
  loading,
  handleCreateUser }) => {
  const [ email, setEmail ] = useState('');
  const [ password, setPassword ] = useState('');
  const [ firstName, setFirstName ] = useState('');
  const [ lastName, setLastName ] = useState('');
  const [ businessName, setBusinessName ] = useState('');
  const [ phone, setPhone ] = useState('');

const handleSubmit = (e) => {
  e.preventDefault();
};

return (    
      <div className="LoginPage"> 
        <form onSubmit={ e => handleSubmit(e)}>
            <label>  
                Email
              <input
                name="email"
                type="email"
                value={email}
                onChange={ e => setEmail( e.target.value ) }
                placeholder="email"
              >
              </input>
            </label>
              <label>
                Password   
                <input
                    name="password"
                    type="password"
                    value={password}
                    onChange={ e => setPassword( e.target.value ) }
                    placeholder="password"
                >
                </input>
              </label>
              <label>
                First Name   
              <input
                  name="firstName"
                  type="text"
                  value={firstName}
                  onChange={ e => setFirstName( e.target.value ) }
                  placeholder="first name"
              >
              </input>
              </label>
              <label>
                Last Name   
                <input
                    name="lastName"
                    type="text"
                    value={lastName}
                    onChange={ e => setLastName( e.target.value ) }
                    placeholder="last name"
                >
                </input>
                </label>
                <span className="LoginPageRadioButton">
                  <span className="left">
                      <label>                         
                        School
                        <input
                          name="firstName"
                          type="radio"
                          value={role.Student}
                         // onChange={ e => setUserRole( e.target.value ) }
                          placeholder="first name"
                        >
                        </input>
                      </label>                    
                  </span>                    
                 <span className="right">
                 <label>
                    Organization
                    <input
                      name="firstName"
                      type="radio"
                      value={role.Tutor}
                      //onChange={ e => setUserRole( e.target.value ) }
                      placeholder="first name"
                    >
                    </input> 
                  </label>  
                 </span>      
                 <span className="right">
                 <label>
                    Individual
                    <input
                      name="firstName"
                      type="radio"
                      value={role.Tutor}
                      //onChange={ e => setUserRole( e.target.value ) }
                      placeholder="first name"
                    >
                    </input> 
                  </label>  
                 </span>                                              
                  </span>
                <label>
                  Business Name   
                  <input
                      name="businessName"
                      type="text"
                      value={businessName}
                      onChange={ e => setBusinessName( e.target.value ) }
                      placeholder="business name"
                  >
                  </input>
                  </label>
                  <label>
                    Phone
                    <input
                        name="phone"
                        type="phone"
                        value={phone}
                        onChange={ e => setPhone( e.target.value ) }
                        placeholder="phone"
                    >
                    </input>
                    </label>
                    <div>                            
                    </div>
                      { error  && (<div className="error"> { error.message }</div>)}
                        <div></div>                  
                      <button
                          type="submit"
                          disabled={loading}
                          onClick={e => handleCreateUser(email, password, firstName, lastName, businessName, phone)}
                      >
                        Create Service Account
                    </button>
            </form>                     
        </div> 
    );
};

export default OperatorRegistrationForm;