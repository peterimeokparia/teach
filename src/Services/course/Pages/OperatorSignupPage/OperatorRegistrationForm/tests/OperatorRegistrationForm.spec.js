import React from 'react';
import OperatorRegistrationForm  from 'services/course/pages/OperatorSignupPage/OperatorRegistrationForm';
import renderer from 'react-test-renderer';

describe('OperatorRegistrationForm', () =>  {  
  beforeEach(() => {
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
        .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {
    const tree = renderer.create(
        <OperatorRegistrationForm /> 
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




