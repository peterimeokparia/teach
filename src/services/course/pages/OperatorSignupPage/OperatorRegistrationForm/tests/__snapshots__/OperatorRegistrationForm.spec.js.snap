// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatorRegistrationForm renders as expected 1`] = `
<div
  className="LoginPage"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <label>
      Password
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
    </label>
    <label>
      First Name
      <input
        name="firstName"
        onChange={[Function]}
        placeholder="first name"
        type="text"
        value=""
      />
    </label>
    <label>
      Last Name
      <input
        name="lastName"
        onChange={[Function]}
        placeholder="last name"
        type="text"
        value=""
      />
    </label>
    <span
      className="LoginPageRadioButton"
    >
      <span
        className="left"
      >
        <label>
          School
          <input
            name="firstName"
            placeholder="first name"
            type="radio"
            value="Student"
          />
        </label>
      </span>
      <span
        className="right"
      >
        <label>
          Organization
          <input
            name="firstName"
            placeholder="first name"
            type="radio"
            value="Tutor"
          />
        </label>
      </span>
      <span
        className="right"
      >
        <label>
          Individual
          <input
            name="firstName"
            placeholder="first name"
            type="radio"
            value="Tutor"
          />
        </label>
      </span>
    </span>
    <label>
      Business Name
      <input
        name="businessName"
        onChange={[Function]}
        placeholder="business name"
        type="text"
        value=""
      />
    </label>
    <label>
      Phone
      <input
        name="phone"
        onChange={[Function]}
        placeholder="phone"
        type="phone"
        value=""
      />
    </label>
    <div />
    <div />
    <button
      onClick={[Function]}
      type="submit"
    >
      Create Service Account
    </button>
  </form>
</div>
`;
